torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -483,7 +483,7 @@ def shard_parameter(
             sharding_spec.dim,  # type: ignore[arg-type]
             local_metadata.shard_offsets[sharding_spec.dim],  # type: ignore[union-attr, arg-type, index]
             local_metadata.shard_sizes[sharding_spec.dim],  # type: ignore[union-attr, index]
-        ).contiguous()
```

Comment:
im wondering can we just do `tensor.detach().clone()` instead of no grad context manager? Also I think for meta tensors it should not be very expensive, but for normal tensor we will suffer from a copy overhead?

Reply:
I guess so. The initial idea behind adding this context manager is to make the local shard a leaf node in the autograd; otherwise, we need to explicitly call `require_grad` for the local shard. So looks like this clone is needed anyway?

Reply:
Synced offline, use `detach()` instead.